test(linter): add `OxlintRules::override_rules` tests

DonIsaac opened the pull request:
wrote these while working on better understanding this method in prep for the `LinterBuilder` migration.

DonIsaac commented:
> ## [CodSpeed Performance Report](https://codspeed.io/oxc-project/oxc/branches/don/09-26-test_linter_add_oxlintrules_override_rules_tests)
> ### Merging #6081 will **degrade performances by 25.26%**
> Comparing `don/09-26-test_linter_add_oxlintrules_override_rules_tests` ([448a000](https://github.com/oxc-project/oxc/commit/448a000c2fb11a4cd9c285508b54a420125f9405)) with `don/09-26-chore_linter_add_filter_parsing_test_cases` ([a23c9a4](https://github.com/oxc-project/oxc/commit/a23c9a42134db3a3beff1a2af856c6f0fc3bd223))
> 
> ### Summary
> `❌ 1` regressions `✅ 28` untouched benchmarks
> 
> > ⚠️ _Please fix the performance issues or [acknowledge them on CodSpeed](https://codspeed.io/oxc-project/oxc/branches/don/09-26-test_linter_add_oxlintrules_override_rules_tests)._
> 
> ### Benchmarks breakdown
> Benchmark	`don/09-26-chore_linter_add_filter_parsing_test_cases`	`don/09-26-test_linter_add_oxlintrules_override_rules_tests`	Change
> ❌	`isolated-declarations[vue-id.ts]`	396.8 ms	530.9 ms	-25.26%

Most likely a fluke, since I didn't touch anything in ID. I'll re-run benchmarks.

camchenry commented:
> > ## [CodSpeed Performance Report](https://codspeed.io/oxc-project/oxc/branches/don/09-26-test_linter_add_oxlintrules_override_rules_tests)
> > ### Merging #6081 will **degrade performances by 25.26%**
> > Comparing `don/09-26-test_linter_add_oxlintrules_override_rules_tests` ([448a000](https://github.com/oxc-project/oxc/commit/448a000c2fb11a4cd9c285508b54a420125f9405)) with `don/09-26-chore_linter_add_filter_parsing_test_cases` ([a23c9a4](https://github.com/oxc-project/oxc/commit/a23c9a42134db3a3beff1a2af856c6f0fc3bd223))
> > ### Summary
> > `❌ 1` regressions `✅ 28` untouched benchmarks
> > > ⚠️ _Please fix the performance issues or [acknowledge them on CodSpeed](https://codspeed.io/oxc-project/oxc/branches/don/09-26-test_linter_add_oxlintrules_override_rules_tests)._
> > 
> > 
> > ### Benchmarks breakdown
> > Benchmark	`don/09-26-chore_linter_add_filter_parsing_test_cases`	`don/09-26-test_linter_add_oxlintrules_override_rules_tests`	Change
> > ❌	`isolated-declarations[vue-id.ts]`	396.8 ms	530.9 ms	-25.26%
> 
> Most likely a fluke, since I didn't touch anything in ID. I'll re-run benchmarks.

agreed, doesn't seem like this would impact perf at all.

DonIsaac committed:
test(linter): add `OxlintRules::override_rules` tests (#6081)

wrote these while working on better understanding this method in prep for the `LinterBuilder` migration.
